central/vuln-mgmt/workload/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,218 @@
+) []*v1.VulnMgmtExportWorkloadsResponse {
+	out, err := client.VulnMgmtExportWorkloads(s.ctx, request)
+	s.Require().NoError(err)
+	var results []*v1.VulnMgmtExportWorkloadsResponse
+	for {
+		chunk, err := out.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		results = append(results, chunk)
+	}
+	return results
+}
+
+func (s *servicePostgresTestSuite) TestExport() {
+	cases := []struct {
+		name        string
+		deployments []*storage.Deployment
+		query       string
+		expected    []*v1.VulnMgmtExportWorkloadsResponse
+	}{
+		{
+			name:        "zero deployments",
+			deployments: []*storage.Deployment{},
+		},
+		{
+			name: "one deployment no query",
+			deployments: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+			},
+			expected: []*v1.VulnMgmtExportWorkloadsResponse{
+				{
+					Deployment: s.createDeployment(fixtureconsts.Deployment1),
+					Images:     fixtures.DeploymentImages(),
+				},
+			},
+		},
+		{
+			name: "multiple deployments no query",
+			deployments: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+				s.createDeployment(fixtureconsts.Deployment2),
+				s.createDeployment(fixtureconsts.Deployment3),
+			},
+			expected: []*v1.VulnMgmtExportWorkloadsResponse{
+				{
+					Deployment: s.createDeployment(fixtureconsts.Deployment1),
+					Images:     fixtures.DeploymentImages(),
+				},
+				{
+					Deployment: s.createDeployment(fixtureconsts.Deployment2),
+					Images:     fixtures.DeploymentImages(),
+				},
+				{
+					Deployment: s.createDeployment(fixtureconsts.Deployment3),
+					Images:     fixtures.DeploymentImages(),
+				},
+			},
+		},
+		{
+			name: "multiple deployments with query",
+			deployments: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+				s.createDeployment(fixtureconsts.Deployment2),
+				s.createDeployment(fixtureconsts.Deployment3),
+			},
+			query: fmt.Sprintf("Deployment ID:%s", fixtureconsts.Deployment2),
+			expected: []*v1.VulnMgmtExportWorkloadsResponse{
+				{
+					Deployment: s.createDeployment(fixtureconsts.Deployment2),
+					Images:     fixtures.DeploymentImages(),
+				},
+			},
+		},
+	}
+	for _, c := range cases {
+		s.T().Run(c.name, func(t *testing.T) {
+			s.upsertDeployments(c.deployments)
+
+			request := &v1.VulnMgmtExportWorkloadsRequest{Timeout: 5, Query: c.query}
+			conn, closeFunc := s.createGRPCWorkloadsService(s.ctx)
+			defer closeFunc()
+			client := v1.NewVulnMgmtWorkloadServiceClient(conn)
+			results := s.receiveWorkloads(client, request)
+
+			s.Require().Len(results, len(c.expected))
+			for i := range results {
+				s.Assert().Equal(c.expected[i].Deployment, results[i].Deployment)
+				s.Require().Len(results[i].Images, len(c.expected[i].Images))
+				for j := range results[i].Images {
+					s.Assert().Equal(c.expected[i].Images[j].GetId(), results[i].Images[j].GetId())
+					s.Assert().Equal(c.expected[i].Images[j].GetId(), results[i].Images[j].GetId())
+				}
```

Comment:
Can you not just do a `assert.ElementsMatch()` here or is there some timestamp that will be added?

Reply:
Yeah the images contain timestamps. I added a comment to clarify.